### ISSUE TITLE
Negative index slicing seems to behave weird on devito.Data

### ISSUE THREAD

**Issue by navjotk** [opesci/devito#1006]
```
grid = Grid(shape=(20,))
u = TimeFunction(name='u', save=None, grid=grid, space_order=0, time_order=1)

print(u.data[1, -2:])
```

produces the error: `IndexError: index 20 is out of bounds for axis 0 with size 2`

but the error does not happen on numpy objects. Also, it only happens when printing.

**Comment by mloubout** [comment]
Looks like `__repr__` is fine but `__str__` is not

**Comment by rhodrin** [comment]
Thanks. Will look into this today.

**Comment by rhodrin** [comment]
So I assume this must have been a problem for a long time then? Basically the 'local indices' don't get passed correctly meaning you get out of bounds errors. This is fixed by adding:
```
    def __str__(self):
        return super(Data, self._local).__str__()
```
I'll include the above in a PR which also fixes #1007.

**Comment by rhodrin** [comment]
Fixed on https://github.com/opesci/devito/tree/misc-data-fixes. Will be PR'ing after adding a few more tests.

**Comment by FabioLuporini** [comment]
closing as fixed by #1008
